modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/FileEntryPicker.tsx

```diff
@@ -0,0 +1,321 @@
+
+		disableFormSubmitButton(false, portletNamespace);
+
+		selectFileButtonRef.current?.focus();
+	};
+
+	const validateFrontendTokenDefinition = async (
+		body: any,
+		endpoint: string
+	) => {
+		try {
+			const response = await fetch(
+				'/o/com-liferay-frontend-token-definition-impl' + endpoint,
+				{
+					body,
+					method: 'POST',
+				}
+			);
+
+			const data = await response.json();
+
+			let displayType: DisplayType = 'success';
+
+			if (!response.ok) {
+				displayType = 'danger';
+
+				disableFormSubmitButton(true, portletNamespace);
+			}
+			else {
+				disableFormSubmitButton(false, portletNamespace);
+			}
+
+			setFeedback({
+				displayType,
+				message: data.message,
+			});
+		}
+		catch (error) {
+			clearSelection();
+
+			setFeedback({
+				displayType: 'danger',
+				message: Liferay.Language.get(
+					'your-upload-failed-to-complete-please-try-again-or-contact-support-if-the-error-persists'
+				),
+			});
+		}
+	};
+
+	const handleFileInputChange = async ({
+		target,
+	}: ChangeEvent<HTMLInputElement>) => {
+		const filePath = target.value;
+
+		const fileName = filePath.replace(/^.*[\\]/, '');
+
+		if (!fileName.endsWith('.json')) {
+			setFeedback({
+				displayType: 'danger',
+				message: Liferay.Language.get(
+					'the-format-is-not-valid-please-upload-a-valid-frontend-token-definition-json-file'
+				),
+			});
+
+			disableFormSubmitButton(true, portletNamespace);
+
+			return;
+		}
+
+		setIsValidatingJSON(true);
+
+		if (!alertRole) {
+			setAlertRole('alert');
+		}
+
+		setFeedback({
+			displayType: 'info',
+			message: Liferay.Language.get(
+				'the-frontend-token-definition-json-file-is-being-uploaded-and-validated'
+			),
+		});
+
+		if (target.files === null) {
+			return;
+		}
+
+		const file = target.files[0];
+
+		try {
+			const frontendTokenDefinitionString = await readInputFile(file);
+
+			await validateFrontendTokenDefinition(
+				objectToFormData({
+					file,
+				}),
+				'/validate-file'
+			);
+
+			setFrontendTokenDefinition(frontendTokenDefinitionString);
+
+			setIsValidatingJSON(false);
+		}
+		catch (error) {
+			clearSelection();
+
+			setFeedback({
+				displayType: 'danger',
+				message: Liferay.Language.get(
+					'the-format-is-not-valid-please-upload-a-valid-frontend-token-definition-json-file'
+				),
+			});
+		}
+	};
+
+	function readInputFile(file: File) {
+		const fileReader = new FileReader();
+
+		return new Promise<string>((resolve, reject) => {
+			fileReader.onerror = () => {
+				fileReader.abort();
+
+				reject(
+					new DOMException(
+						Liferay.Language.get(
+							'your-upload-failed-to-complete-please-try-again-or-contact-support-if-the-error-persists'
+						)
+					)
+				);
+			};
+
+			fileReader.onload = () => {
+				resolve(fileReader.result as string);
+			};
+
+			fileReader.readAsText(file);
+		});
+	}
+
+	useEffect(() => {
+		if (frontendTokenDefinition && externalReferenceCode) {
+			validateFrontendTokenDefinition(
```

Comment:
Most of what happens in validateFrontendTokenDefinition doesn't seem applicable to calling tokens-info. We shouldn't ever disable the submit button. The response should always be okay. There are errors in there related to uploading, which doesn't happen for this api call. I think too much of this is shared where it doesn't actually make sense.